_posts/2019-03-18-rules-jvm-external-maven.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: posts
+title: "A repository rule for calculating transitive Maven dependencies"
+authors:
+  - jin
+  - dkelmer
+---
+
+We’re delighted to introduce the initial release of
+[`rules_jvm_external`](https://github.com/bazelbuild/rules_jvm_external), a
+repository rule to resolve and fetch artifacts transitively in Maven
+repositories, like Maven Central and Google Maven. This project is officially
+maintained by the Bazel team.
+
+## Background and motivation
+
+Maven integration in Bazel is a large space with many existing robust solutions
+from community members:
+
+* [https://github.com/johnynek/bazel-deps](https://github.com/johnynek/bazel-deps)
+    * Tool to generate Bazel dependencies transitively for Maven artifacts, with
+      Scala and Kotlin support
+* [https://github.com/square/bazel\_maven\_repository](https://github.com/square/bazel_maven_repository)
+    * Repository rule to create an idiomatic Bazel representation of a Maven
+      repository using a pinned list of artifacts, with support for Kotlin and
+      Android dependencies
+* [https://github.com/menny/mabel](https://github.com/menny/mabel)
+    * Tool to generate a Maven dependency graph as a lock file, with support for
+      Kotlin and Android dependencies
+
+Each project approaches the problem slightly differently. Check out those
+projects as well to see their full feature set and determine which solution
+works best for you. It is also possible to wire dependencies up manually using
+rules like `java_import_external` and `aar_import_external`.
+
+## Objectives
+
+We wanted to fulfill two primary objectives with `rules_jvm_external`.
+
+The first is that users should be able to run `bazel build //...`. This means
+that the calculation of the transitive dependencies and their fetching need to
+happen during repository rule execution time. This makes dependency fetching
+transparent to users and reduces onboarding friction to a Bazel-built project.
+
+The second is to provide an interface that is familiar to users with projects
```

Comment:
"Second, we wanted to provide..." maintains parallel structure with the previous change.